### ISSUE TITLE
[Bug]: BrowserContexts.close() cause chromium quit?

### ISSUE THREAD

**Issue by yyx1111** [puppeteer/puppeteer#8691]
### Bug description

let context = await browser.createIncognitoBrowserContext();
let page = await context.newPage();
let context1 = await browser.createIncognitoBrowserContext();
let page1 = await context1.newPage();
await context1.close()

///////////////////////////////
chromium quit?
browser.isConnected()==true?

### Puppeteer version

15.5.0

### Node.js version

16.16.0

### npm version

8.11.0

### What operating system are you seeing the problem on?

Windows

**Comment by OrKoN** [comment]
I am unable to reproduce with the following script:

```
const puppeteer = require('puppeteer');

(async () => {
    const browser = await puppeteer.launch();
    let context = await browser.createIncognitoBrowserContext();
    let page = await context.newPage();
    let context1 = await browser.createIncognitoBrowserContext();
    let page1 = await context1.newPage();
    await context1.close();
    await new Promise(resolve => setTimeout(resolve, 1000))
    console.log(browser.isConnected());
})();
```

browser.isConnected() reports true which appears to be expected. Please provide more details about the issue.

**Comment by yyx1111** [comment]
(async () => {
  const browser = await puppeteer.launch({ headless: false });
  let context = await browser.createIncognitoBrowserContext();
  let page = await context.newPage();
  let context1 = await browser.createIncognitoBrowserContext();
  let page1 = await context1.newPage();
  await context1.close();
  await new Promise((resolve) => setTimeout(resolve, 1000));
  console.log(browser.isConnected()); //true,chromium quit
})();

**Comment by Kikobeats** [comment]
I created a repo that reproduces the issue:
https://github.com/Kikobeats/puppeteer-8691

**Comment by Kikobeats** [comment]
I created a repo that reproduces the issue:
https://github.com/Kikobeats/puppeteer-8691

**Comment by Kikobeats** [comment]
I run different puppeteer/chrome setup to try to find the latest build working:

```
| puppeteer | chrome              | status |
|-----------|---------------------|--------|
| 16.1.0    | Chrome/105.0.5173.0 | ❌      |
| 16.0.0    | Chrome/105.0.5173.0 | ❌      |
| 15.5.0    | Chrome/105.0.5173.0 | ❌      |
| 15.4.2    | Chrome/104.0.5109.0 | ✅      |
```

It looks like could be a specific problem related with Chromium.

@OrKoN not sure how to but could be possible test it against v106?

**Comment by OrKoN** [comment]
@Kikobeats yeah, it seems it is likely a Chromium issue. You can modify the revision in src/revisions.ts to test a different binary with the same version. We also need to include this scenario into our tests.

**Comment by stevenwdv** [comment]
I also have this issue. Here's a repro which seems to be more reliable for me:
```js
import puppeteer from 'puppeteer';
import {setTimeout} from 'node:timers/promises';

const browser = await puppeteer.launch({headless: false});

const ctxs = await Promise.all(Array(3).fill(null).map(async () => {
	const ctx = await browser.createIncognitoBrowserContext();
	await ctx.newPage();
	return ctx;
}));
for (const [i, ctx] of ctxs.entries()) {
	await setTimeout(1_000);
	console.log(`closing ${i}`);
	await ctx.close();
	console.log(`closed ${i}`);
}
await setTimeout(1_000);
await browser.close();
console.info('all done');
```

Output:
```
closing 0
closed 0
closing 1
file:///C:/dev/puppeteer-test/node_modules/puppeteer/lib/esm/puppeteer/common/Connection.js:101
                error: new ProtocolError(),
                       ^

ProtocolError: Protocol error (Target.disposeBrowserContext): Target closed.
    at file:///C:/dev/puppeteer-test/node_modules/puppeteer/lib/esm/puppeteer/common/Connection.js:101:24
    at new Promise (<anonymous>)
    at Connection.send (file:///C:/dev/puppeteer-test/node_modules/puppeteer/lib/esm/puppeteer/common/Connection.js:97:16)      
    at Browser._disposeContext (file:///C:/dev/puppeteer-test/node_modules/puppeteer/lib/esm/puppeteer/common/Browser.js:296:70)
    at BrowserContext.close (file:///C:/dev/puppeteer-test/node_modules/puppeteer/lib/esm/puppeteer/common/Browser.js:667:74)   
    at file:///C:/dev/puppeteer-test/main.mjs:14:12 {
  originalMessage: ''
}
```

I checked, and this issue occurs with puppeteer v16.2.0 with bundled Chromium v105.0.5173.0 but **not** with the same puppeteer version but Chrome v104.0.5112.102, so it seems to be an upstream issue.

**Comment by stevenwdv** [comment]
Should I file a Chromium bug for this?

**Comment by OrKoN** [comment]
@stevenwdv it'd be great if you could!

**Comment by stevenwdv** [comment]
Bug at https://crbug.com/1358933

**Comment by stevenwdv** [comment]
Note: this still happens with Chromium 106.0.5249.0 from puppeteer 17.1.0

**Comment by stevenwdv** [comment]
Nice, fixed in 18.1.0 as expected!

**Comment by heaven** [comment]
I'm on 19 and the problem is still there.
